lib/ansible/module_utils/network/ios/config/interfaces/interfaces.py

```diff
@@ -105,7 +105,10 @@ def set_state(self, want, have):
                   to the deisred configuration
         """
         commands = []
+
         state = self._module.params['state']
+        if state in ('overridden', 'merged', 'replaced') and not want:
```

Comment:
Note: It doesn't matter here but if you're ever in a performance critical part of the code, reversing the order of these two tests will usually be faster.  The state check has to construct the tuple and then iterate over the tuple looking for state within it whereas the check for want is a simpler truth test.